torch/csrc/Module.cpp

```diff
@@ -744,6 +746,17 @@ PyObject* initModule() {
   // This reference is meant to be given away, so no need to incref here.
   ASSERT_TRUE(set_module_attr("default_generator", (PyObject*)THPDefaultCPUGenerator, /* incref= */ false));
 
+  auto num_gpus = at::detail::getCUDAHooks().getNumGPUs();
+  auto default_cuda_generators = PyList_New(static_cast<Py_ssize_t>(num_gpus));
+  for(int i = 0; i < num_gpus; i++) {
+    auto gen = at::detail::getCUDAHooks().getDefaultCUDAGenerator(i);
+    auto cast_gen = (THPGenerator*)THPGenerator_initDefaultGenerator(gen);
+    // This reference is meant to be given away, so no need to incref here.
+    PyList_SetItem(default_cuda_generators, i, (PyObject*)cast_gen);
+  }
+  // This reference is meant to be given away, so no need to incref here.
+  ASSERT_TRUE(set_module_attr("default_cuda_generators", default_cuda_generators, /* incref= */ false));
```

Comment:
One danger here: `default_cuda_generators` being a list suggests you can assign to it, but assigning to it doesn't actually do anything (in particular, it doesn't overwrite the previous default generator.)

Reply:
Oh interesting. I just tried the following. Is the following behavior correct/desirable? Looks torch.default_generator can be overwritten?
```
Python 3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:22:34) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> torch.default_generator
<torch._C.Generator object at 0x7f9a13452790>
>>> torch.default_generator = torch.Generator()
>>> torch.default_generator
<torch._C.Generator object at 0x7f9a078f1070>
>>> torch.default_cuda_generators
[<torch._C.Generator object at 0x7f9a134527d0>, <torch._C.Generator object at 0x7f9a134527f0>, <torch._C.Generator object at 0x7f9a13452810>, <torch._C.Generator object at 0x7f9a13452830>]
>>> torch.default_cuda_generators = []
>>> torch.default_cuda_generators
[]
>>> 
```

Reply:
technically these should all be un-writable descriptors, but I don't know how easy it is to achieve that from python C api.